geomstats/backend/tensorflow/__init__.py

```diff
@@ -70,57 +78,211 @@ def get_mask_float(indices, mask_shape):
-    TODO(pchauchat): give shape of the output according to guideline
+    tf_mask : array, shape=[mask_shape[:axis], n_samples, mask_shape[axis:]]
     """
-    mask = get_mask_float(indices, mask_shape)
+    mask = get_mask_float(indices, mask_shape, dtype)
     return duplicate_array(mask, n_samples, axis=axis)
 
 
-def assignment_single_value(x, value, indices, axis=0):
+def assignment_single_value_by_sum(x, value, indices, axis=0):
+    """Add a value at given indices of an array.
+
+    Parameters
+    ----------
+    x: array-like, shape=[dimension]
+        Initial array.
+    value: float
+        Value to be added.
+    indices: {int, tuple, list(int), list(tuple)}
+        Single int or tuple, or list of ints or tuples of indices where value
+        is assigned.
+        If the length of the tuples is shorter than ndim(x), value is
+        assigned to each copy along axis.
+    axis: int, optional
+        Axis along which value is assigned, if vectorized.
+
+    Returns
+    -------
+    x_new : array-like, shape=[dimension]
+        Copy of x where value was added at all indices (and possibly along
+        an axis).
+    """
     single_index = not isinstance(indices, list)
     if single_index:
         indices = [indices]
-    use_vectorization = (len(indices[0]) < ndim(x))
+    if isinstance(indices[0], tuple):
+        use_vectorization = (len(indices[0]) < ndim(x))
+    else:
+        use_vectorization = ndim(x) > 1
 
     if use_vectorization:
         n_samples = shape(x).numpy()[0]
         mask = get_vectorized_mask_float(
-            n_samples, indices, shape(x).numpy()[1:], axis)
+            n_samples, indices, shape(x).numpy()[1:], axis, x.dtype)
     else:
-        mask = get_mask_float(indices, shape(x))
-    x = x + value * mask
-    return x
+        mask = get_mask_float(indices, shape(x), x.dtype)
+    x_new = x + value * mask
+    return x_new
+
+
+def assignment_by_sum(x, values, indices, axis=0):
+    """Add values at given indices of an array.
+
+    Parameters
+    ----------
+    x: array-like, shape=[dimension]
+        Initial array.
+    values: {float, list(float)}
+        Value or list of values to be assigned.
+    indices: {int, tuple, list(int), list(tuple)}
+        Single int or tuple, or list of ints or tuples of indices where value
+        is assigned.
+        If the length of the tuples is shorter than ndim(x), values are
+        assigned to each copy along axis.
+    axis: int, optional
+        Axis along which values are assigned, if vectorized.
+
+    Returns
+    -------
+    x_new : array-like, shape=[dimension]
+        Copy of x as the sum of x and the values at the given indices.
+
+    Notes
+    -----
+    If a single value is provided, it is assigned at all the indices.
+    If a list is given, it must have the same length as indices.
+    """
+    if not isinstance(values, list):
+        return assignment_single_value_by_sum(x, values, indices, axis)
+
+    else:
```

Comment:
This else is not tested.

Reply:
I had added additonal tests on an awaiting PR. Do I bring them on this one ?

Reply:
Let's wait for the other PR then!